altair/api.py

```diff
@@ -115,6 +117,35 @@ def to_altair(self, data=None):
         """Emit the Python code as a string required to created this Chart."""
         return str(self._to_code(data=data))
 
+    def _cleanup_data(self):
+        """Remove unused data columns in-place.
+
+        This method must be called *after* self._finalize()
```

Comment:
Not quite sure this is true. Do you mean "If this method is called, it must be *after* self._finalize()" In other words, not calling this method is fine, but if you do has to be after.